python/metatensor-learn/metatensor/learn/_classes.py

```diff
@@ -0,0 +1,28 @@
+# This file defines the default set of classes to use in metatensor-learn
+#
+# The code from metatensor-learn can be used in two different modes: either pure Python
+# mode or TorchScript mode. In the second case, we need to use a different version of
+# the classes above; to do so without having to rewrite everything, we re-import
+# metatensor-learn in a special context.
+#
+# See metatensor-torch/metatensor/torch/learn.py for more information.
+#
+# Any change to this file MUST be also be made to `metatensor/torch/learn.py`.
+
+from metatensor import Labels, LabelsEntry, TensorBlock, TensorMap
+
+
+def torch_jit_is_scripting():
+    return False
+
+
+check_isinstance = isinstance
+
+__all__ = [
+    "Labels",
+    "TensorBlock",
+    "TensorMap",
+    "LabelsEntry",
```

Comment:
I still don't understand why `LabelsEntry` is here. Are you explicitly manipulating it? This class should be mostly hidden from the user.

Reply:
it is used as key for the ModuleMap

Reply:
This won't work: we can not define a custom `__hash__` in TorchScript, meaning `__hash__` uses `id()`, so something like this will fail:

```py
from metatensor.torch import Labels

labels = Labels.range("a", 3)

data = {
    entry: 0 for entry in labels
}
print(data) # {LabelsEntry(a=0): 0, LabelsEntry(a=1): 0, LabelsEntry(a=2): 0}

other = Labels.range("a", 3)

for entry in other:
     data[entry] # throw KeyError: LabelsEntry(a=0)
```

#303 is related.

Reply:
We agreed on using Labels.position #427